### DISCUSSION TITLE
Output Parsing Error and Incomplete User Information Retrieval

### DISCUSSION THREAD

**Discussion by JayyShah** [stepanogil/autonomous-hr-chatbot#3] [General]
I encountered an issue while using the chatbot. I attempted to ask the same questions as shared in the screenshot . However, I consistently received an "Output Parsing Error" when requesting specific information.

Additionally, when I asked the chatbot "Who am I and what is my employee ID?", it only provided my employee ID without mentioning my name. This incomplete identification of the user poses a limitation in personalization and could hinder the overall user experience.

**Comment by stepanogil** [comment]
thank you for your feedback.
This is a known issue - LC agent has an output parser that expects an output to be formatted in a certain way for all requests (Thought/Action/Action Input...) You'll run into this error if the LLM decides to be ornery. Try doing any of the following:

1. try submitting the request again. 
2. add additional few shot examples in the tool description 
 e.g. user: what's my name and employee id
assistant: df[df[name == {user}][['name', 'employee_id']]
3. try creating a tool using the newly released function calling feature by OpenAI. those models are fine tuned to follow the expected format by the function call (json/key-value pair)
4. try using the GPT-4 api if you have access to it.


Thanks
Stephen
